easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,158 @@
+##
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Code_Saturne, implemented as an easyblock
+
+@author: Metin Cakircali (Juelich Supercomputing Centre)
+"""
+
+import os
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.run import run_cmd
+from easybuild.tools.filetools import apply_regex_substitutions
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Code_underscore_Saturne(EasyBlock):
+    """Support for building and installing Code_Saturne."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters specific to Code_Saturne."""
+        extra_vars = EasyBlock.extra_options(extra_vars)
+
+        # add more custom easyconfig parameters specific to OpenFOAM
+        extra_vars.update({
+            'debug': [False, "Build the debug version.", CUSTOM],
+            'slurm': [False, "Build for the slurm resource manager.", CUSTOM],
```

Comment:
The default doesn't create a valid `.cfg` file. Also, using slurm doesn't automatically warrant the use of `srun`. I suggest to remove this variable but replace it with 
```
            'batch': ['SLURM', "Build for the slurm resource manager/scheduler.", CUSTOM],
            'mpiexec': ['srun', "Command to run MPI jobs", CUSTOM],
            'mpiexec_n': ['-n', "Flag to specify number of MPI tasks", CUSTOM],
            'mpiexec_n_per_node': ['--ntasks-per-node', "Flag to specify number of MPI tasks per node.", CUSTOM],
```

Reply:
thx for the example. I want to keep it simple. Of course, should be flexible enough too.
Is there a situation/system that srun doesn't work with SLURM?

Reply:
Well, I don't know every system in the world, but using `srun` is not a requirement to use SLURM. LRZ for instance has mpiexec in their documentation: https://www.lrz.de/services/compute/linux-cluster/batch_parallel/

Reply:
I know it's not requirement, but it doesn't mean that srun wouldn't work.. They also refer to srun in their documentation. I don't see a reason to complicate, that's all.